Sources/Workspace/Workspace.swift

```diff
@@ -1994,6 +1996,18 @@ extension Workspace {
             }
         }
 
+        // Look for any cycle in the dependencies.
+        if let cycle = try findCycle(topologicalSortInput, successors: topologicalSortSuccessors) {
+            observabilityScope.emit(
+                error: "cyclic dependency declaration found: " +
+                (cycle.path + cycle.cycle).map({ $0.key.identity.description }).joined(separator: " -> ") +
+                " -> " + cycle.cycle[0].key.identity.description
+            )
+            // return partial results
+            return DependencyManifests(root: root, dependencies: [], workspace: self)
+        }
+        let allManifestsWithPossibleDuplicates = try topologicalSort(topologicalSortInput, successors: topologicalSortSuccessors)
```

Comment:
I think I'd prefer that we caught the error of `topologicalSort()` and tried to find cycles post hoc. Otherwise, we're going to do a bunch of extra graph traversal here.

Reply:
I agree with you in principal, but this is the existing pattern in the code base, so I did not want to do a larger refactoring. we should record this however as a follow on item